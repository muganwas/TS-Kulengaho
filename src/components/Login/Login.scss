.login-component-container {
  padding: 5px;
}
