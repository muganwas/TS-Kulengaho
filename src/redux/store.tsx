import { combineReducers, applyMiddleware, createStore } from 'redux';
import thunk from 'redux-thunk';
import promise from 'redux-promise-middleware';
import { apiMiddleware } from 'redux-api-middleware';
import logger from 'redux-logger';

var middleware;
// apply logger only in development and staging
if (process.env.NODE_ENV !== 'production') middleware = applyMiddleware(promise, thunk, apiMiddleware, logger);
else middleware = applyMiddleware(promise, thunk, apiMiddleware);

const initialState = {};

const allReducers = combineReducers({
//put all reducers here
});

const store = createStore(
    allReducers,
    initialState,
    middleware 
)

export default store;