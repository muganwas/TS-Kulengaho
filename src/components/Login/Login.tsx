import React from 'react'
import { Dispatch } from 'redux'
import { useSelector, useDispatch } from 'react-redux'
import { TextInput } from '../Form'
import { updateUsername, updateUserPassowrd } from '../../redux/actions'
import { generic_input_func, RootState } from '../../sharedTypes'
import './Login.scss'

const Login: React.FC = () => {
  const dispatch: Dispatch<any> = useDispatch()
  const userInfo = useSelector((store: RootState) => store.userInfo)
  const usernameChange: generic_input_func = (e) => {
    e.preventDefault()
    const id = e.target.id
    const value = e.target.value
    if (id === 'user-name' && userInfo.user_name !== value)
      dispatch(updateUsername(value))
    else if (id === 'password' && userInfo.user_password !== value)
      dispatch(updateUserPassowrd(value))
  }
  return (
    <div
      data-testid='login-component-container'
      className='login-component-container'
    >
      <TextInput
        id='user-name'
        label='Username'
        testid='username-login'
        placeholder='Username'
        onBlur={usernameChange}
      />
      <TextInput
        id='password'
        label='Password'
        inputType='password'
        testid='password-login'
        placeholder='Password'
        onBlur={usernameChange}
      />
    </div>
  )
}

export default Login
