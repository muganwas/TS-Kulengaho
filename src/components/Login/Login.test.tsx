import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import store from '../../redux/store'
import Login from './Login'

describe('Login component renders', () => {
  it('renders login components', () => {
    const { getByTestId, container } = render(
      <Provider store={store}>
        <Login />
      </Provider>
    )
    const containerElement = getByTestId('login-component-container')
    const usernameInputElement = getByTestId('username-login')
    const passwordInputElement = getByTestId('password-login')
    expect(passwordInputElement).toBeInTheDocument()
    expect(usernameInputElement).toBeInTheDocument()
    expect(containerElement).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
