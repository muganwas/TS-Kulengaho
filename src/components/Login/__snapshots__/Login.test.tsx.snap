// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders renders login components 1`] = `
<div>
  <div
    class="login-component-container"
    data-testid="login-component-container"
  >
    <div
      class="form-container"
    >
      <form
        data-testid="login-form"
        id="login-form"
      >
        <div
          class="text-input-container"
        >
          <label>
            Username
          </label>
          <input
            class="text-input "
            data-testid="username-login"
            id="user-name"
            placeholder="Username"
            type="string"
            value=""
          />
          
        </div>
        <div
          class="text-input-container"
        >
          <label>
            Password
          </label>
          <input
            class="text-input "
            data-testid="password-login"
            id="password"
            placeholder="Password"
            type="password"
            value=""
          />
          
        </div>
        <div
          class="button-container"
        >
          <button
            class="button"
            data-testid="login-submit"
            disabled=""
            id="submit-button"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
